Mathlib/MeasureTheory/Measure/WithDensityFinite.lean

```diff
@@ -0,0 +1,294 @@
+/-
+Copyright (c) 2024 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import Mathlib.MeasureTheory.Decomposition.RadonNikodym
+
+/-!
+# s-finite measures can be written as `withDensity` of a finite measure
+
+If `μ` is an s-finite measure, then there exists a finite measure `μ.toFinite` and a measurable
+function `densityToFinite μ` such that `μ = μ.toFinite.withDensity μ.densityToFinite`. If `μ` is
+zero this is the zero measure, and otherwise we can choose a probability measure for `μ.toFinite`.
+
+That measure is not unique, and in particular our implementation leads to `μ.toFinite ≠ μ` even if
+`μ` is a probability measure.
+
+## Main definitions
+
+In all these definitions and the results below, `μ` is an s-finite measure (`SFinite μ`).
+
+* `MeasureTheory.Measure.toFinite`: a finite measure with `μ ≪ μ.toFinite` and `μ.toFinite ≪ μ`.
+  If `μ ≠ 0`, this is a probabilty measure.
+* `MeasureTheory.Measure.densityToFinite`: a measurable function such that
+  `μ = μ.toFinite.withDensity μ.densityToFinite`.
+* `MeasureTheory.Measure.toSigmaFinite`: a sigma-finite measure with `μ ≪ μ.toSigmaFinite` and
```

Comment:
Those two sets and the properties of the restrictions (sigma-finite for one, 0/infty for the other) are the content of my next planned PR. Sorry for not having an overview PR for everything. I am not very organized with these contributions.

I agree that `toSigmaFinite` might not be useful. I defined it while writing that code without thinking much about its uses, perhaps only because it felt like a natural thing to define. I'll delete it.